server/stt/stt_planning_ml.py

```diff
@@ -124,7 +136,221 @@ async def get_coverage_details(
+        planning_elt = news_coverage_elt.find(self.qname("planning"))
+        if planning_elt is None:
+            return
+
+        has_stt_fields = False
+
+        # Check for STT-specific namespaces and qcodes
+        for subj in planning_elt.findall(self.qname("subject")):
+            qcode = subj.get("qcode", "")
+            if any(
+                [
+                    qcode == "sttinternaltext",
+                    qcode == "sttentryinfo",
+                    qcode.startswith("sttworkstatus:"),
+                    qcode.startswith("sttphotoaware:"),
+                    qcode.startswith("sttimagetypename:"),
+                ]
+            ):
+                has_stt_fields = True
+                break
+
+        # Check for internal description fields
+        if not has_stt_fields:
+            for defn in planning_elt.findall(self.qname("definition")):
+                role = defn.get("role", "")
+                if role in ("sttdescription:imagetype", "sttdescription:imagetarget"):
+                    has_stt_fields = True
+                    break
+
+        if not has_stt_fields:
+            if (
+                planning_elt.find(".//{http://www.stt.fi/internal}workstartdate")
+                is not None
+            ):
+                has_stt_fields = True
+
+        if not has_stt_fields:
+            return
+
+        # Parse headline
+        headline_elt = planning_elt.find(self.qname("headline"))
+        if headline_elt is not None and headline_elt.text:
+            coverage.setdefault("planning", {})["headline"] = headline_elt.text.strip()
+
+        # Parse scheduled/due date
+        workstartdate_elt = planning_elt.find(
+            ".//{http://www.stt.fi/internal}workstartdate"
+        )
+        if (
+            workstartdate_elt is not None
+            and workstartdate_elt.text
+            and not coverage.get("planning", {}).get("scheduled")
+        ):
+            try:
+                coverage.setdefault("planning", {})["scheduled"] = self.datetime(
+                    workstartdate_elt.text
+                )
+            except (ValueError, TypeError):
+                logger.warning(
+                    f"Failed to parse workstartdate: {workstartdate_elt.text}"
+                )
+
+        # Parse other STT-specific fields
+        self.parse_coverage_status(planning_elt, coverage)
+        self.parse_coverage_internal_note(planning_elt, coverage)
+        self.parse_picture_type(planning_elt, coverage)
+        self.parse_photographer_awareness(planning_elt, coverage)
+        self.parse_finnish_text_fields(planning_elt, coverage)
+        self.parse_registration_info(planning_elt, coverage)
+
+    def parse_coverage_status(self, planning_elt: Element, coverage: Dict[str, Any]):
+        """Parse coverage status from sttworkstatus subject"""
+        for subject_elt in planning_elt.findall(self.qname("subject")):
```

Comment:
Looks like we're iterating the `subject` fields many times, 1 for each type of data we're populating.
An improvement might be to traverse the `subject` fields once, and call the appropriate `parse_` function based on node attributes.

i.e.
```python
for subject_elt in planning_elt.findall(self.qname("subject")):
    qcode = subject_elt.get("qcode", "")
    if qcode.startswith("sttworkstatus:"):
        parse_coverage_status(subject_elt, coverage)
    elif qcode == "sttinternaltext":
        parse_coverage_internal_note(subject_elt, coverage)
    ...
```